sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -1327,6 +1333,14 @@ CosmosClientBuilder configs(Configs configs) {
         return this;
     }
 
+    private boolean isEmulatorHost(URI uri) {
+        if (StringUtils.isNotEmpty(Configs.getEmulatorHost())) {
+            return Configs.getEmulatorHost().equals(uri.getHost());
+        }
+
+        return "localhost".equals(uri.getHost()) || "::1".equals(uri.getHost()) || "127.0.0.1".equals(uri.getHost());
```

Comment:
"0:0:0:0:0:0:0:1" also valid ipv6 for localhost

Reply:
I think the hostname comparisons shoudl be case-insensitive because the hostname part of Http-uris is always case-insensitive